chromium_src/chrome/browser/ui/webui/searchbox/realbox_handler.cc

```diff
@@ -15,7 +17,32 @@
   autocomplete_input.set_prefer_keyword(true)
 #define set_allow_exact_keyword_match(allow) set_allow_exact_keyword_match(true)
 
+// Unfortunately, plumbing through the source doesn't seem trivial - it looks
+// like it should be possible with the {source} part but it seems like it only
+// works with Google Search. Additionally, the {source} param treats the NTP
+// Realbox and the Omnibox the same, which isn't ideal, so we'd have to patch
+// the behavior there too. This seems like the patch of least changes:
+// 1. If this is a keyword search with Brave Search
+// 2. Then replace &source=desktop with &source=new_tab
+#define OpenURL(params, callback)                                             \
+  GetURL();                                                                   \
```

Comment:
Why are you calling `GetURL` and doing nothing with it? I think any change to omnibox behavior needs a security review cc @brave/sec-team and in this case it also needs a privacy review cc @brave/privacy-review-team and I don't see one attached, but this does not seem like the right place to do this anyway. This looks like it should be done with a browser process URLLoaderThrottle (which still requires sec and privacy review)

Reply:
oic, `GetURL` is for the override? Just fyi for the future that is not a good way to override. `#define OpenURL(PARAMS, CALLBACK) OpenURL(MaybeOverrideURLParams(PARAMS), CALLBACK)` where `MaybeOverrideURLParams` is a method you defined here would be much better, but as I mentioned above we shouldn't be doing this here anyway

Reply:
cc @brave/chromium-src-reviewers ^

Reply:
Cool, that's a much better way of overriding, thanks! 

FWIW, this isn't a change to the `Omnibox` - this is a change to the `Realbox` which is a wrapper around the `Autocomplete` and editing logic which is used by an `Omnibox-like` thing on the NTP. This will have no effect on the Omnibox

Reply:
I see, I misunderstood the issue because it looked like it was supposed to be affecting the omnibox.